main/main.c

```diff
@@ -346,8 +373,11 @@ main(int argc, char **argv)
 	 * after blocks have had a chance to make use of network state
 	 * during stage 2. This is needed because there is no stage 3 for
 	 * the network configuration.
+	 *
+	 * Finalize any user configuration, such as changing file permissions
+	 * and dropping user privileges.
 	 */
-	ret = launch_at_stage2(finalize_stage2, NULL);
+	ret = launch_at_stage2(finalize_stage2, (void *)(intptr_t)log_fd);
```

Comment:
Can't we drop `(intptr_t)` from the cast?

Reply:
If ``(intptr_t)`` was dropped, the error message is `` error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]``